scripts/upgrade

```diff
@@ -89,177 +105,139 @@ if [[ $migration_process -eq 1 ]]; then
+ynh_script_progression --message="Upgrading NGINX web server configuration..."
 
-ynh_script_progression --message="Upgrading database and sources..." --weight=6
+# Create a dedicated NGINX config
+ynh_add_nginx_config
 
-# Before the version 1.7 the upstream version was not stored
-# The way to find the version for the install < 1.7 is to parse the binary file to find which version is installed
-if [ -z ${upstream_version:-} ]; then
-    for version in "0.0." "1.0." "1.1." "1.2." "1.3." "1.4." "1.5." "1.6." "1.7."; do
-        if strings $final_path/gitea | grep -P "^${version//./\\.}\d"; then
-            upstream_version="${version}0"
-            break
-        fi
-    done
-fi
+#=================================================
+# SPECIFIC UPGRADE
+#=================================================
+# UPDATE A CONFIG FILE
+#=================================================
+ynh_script_progression --message="Updating a configuration file..."
 
-restart_gitea() {
-    # Set permissions
-    set_permission
-    ynh_systemd_action -l "Starting new server: tcp:127.0.0.1:" -p "/var/log/$app/gitea.log" -t 30
-    # Leave the time to update the database schema
-    sleep 5
-    systemctl stop $app
-}
+# Clean template to fix issue : https://github.com/gogits/gogs/issues/4585
+ynh_secure_remove --file="$final_path/templates"
 
-case $upstream_version in
-"0.0."* )
-    ynh_setup_source $final_path source/${architecture}_1.0
-    set_permission
-    systemctl start $app
-    sleep 20
-    systemctl stop $app
-;&
-"1.0."* )
-    ynh_setup_source $final_path source/${architecture}_1.1
-    restart_gitea
-;&
-"1.1."* )
-    ynh_setup_source $final_path source/${architecture}_1.2
-    restart_gitea
-;&
-"1.2."* )
-    ynh_setup_source $final_path source/${architecture}_1.3
-    restart_gitea
-;&
-"1.3."* )
-    ynh_setup_source $final_path source/${architecture}_1.4
-    restart_gitea
-;&
-"1.4."* )
-    ynh_setup_source $final_path source/${architecture}_1.5
-    restart_gitea
-;&
-"1.5."* )
-    ynh_setup_source $final_path source/${architecture}_1.6
-    restart_gitea
-;&
-"1.6."* )
-    ynh_setup_source $final_path source/${architecture}_1.7
-    restart_gitea
-;&
-"1.7."* )
-    ynh_setup_source $final_path source/${architecture}_1.8
-    restart_gitea
-;&
-"1.8."* )
-    ynh_setup_source $final_path source/${architecture}_1.9
-    restart_gitea
-;&
-"1.9."* )
-    ynh_setup_source $final_path source/${architecture}_1.10
-    restart_gitea
-;&
-"1.10."* )
-    ynh_setup_source $final_path source/${architecture}_1.11
-    restart_gitea
-;&
-"1.11."* )
-    ynh_setup_source $final_path source/${architecture}_1.12
-    restart_gitea
-;&
-"1.12."* )
-    ynh_setup_source $final_path source/${architecture}_1.13
-    restart_gitea
-;&
-"1.13."* )
-    ynh_setup_source $final_path source/${architecture}_1.14
-    restart_gitea
-;&
-"1.14."* )
-    ynh_setup_source $final_path source/${architecture}_1.15
-    restart_gitea
-;&
-esac
```

Comment:
Well this migration is required. Because as described in the official documentation. If we have actually the version 1.11 and we install the version 1.15 we can't jump between theses main version (because we need to migrate the database) !! We need to install the version 1.12 and start it. Install the version 1.13 and start it and install the version 1.14 and start it before to install the version 1.15.

So please don't remove the code that you don't think it's not needed !!

Reply:
boring... I though it was able to upgrade from one version to the last one